Unit tests: Could not initialize class leakcanary.AppWatcher

### Description

Unit tests fail with 

```
java.lang.ExceptionInInitializerError
	at leakcanary.AppWatcher.isInstalled(AppWatcher.kt:97)
	at leakcanary.AppWatcher.<clinit>(AppWatcher.kt:63)
	at com.kolibree.android.sdk.core.LeakCanaryExtensionsKt.detectLeaks(LeakCanaryExtensions.kt:16)
	at com.kolibree.android.sdk.BluetoothStateReceiver$Companion.onAllConnectionObserversUnsubscribed(BluetoothStateReceiver.kt:53)
	at com.kolibree.android.sdk.BluetoothStateReceiver$Companion.access$onAllConnectionObserversUnsubscribed(BluetoothStateReceiver.kt:31)
	at com.kolibree.android.sdk.BluetoothStateReceiver$Companion$bluetoothStatusObservable$1$1.run(BluetoothStateReceiver.kt:43)
	at io.reactivex.internal.operators.observable.ObservableDoFinally$DoFinallyObserver.runFinally(ObservableDoFinally.java:142)
	at io.reactivex.internal.operators.observable.ObservableDoFinally$DoFinallyObserver.dispose(ObservableDoFinally.java:98)
	at io.reactivex.internal.operators.observable.ObservableDoOnEach$DoOnEachObserver.dispose(ObservableDoOnEach.java:79)
	at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124)
	at io.reactivex.internal.operators.observable.ObservablePublishAlt$PublishConnection.dispose(ObservablePublishAlt.java:161)
	at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124)
	at io.reactivex.internal.operators.observable.ObservableRefCount.timeout(ObservableRefCount.java:156)
	at io.reactivex.internal.operators.observable.ObservableRefCount.cancel(ObservableRefCount.java:103)
	at io.reactivex.internal.operators.observable.ObservableRefCount$RefCountObserver.dispose(ObservableRefCount.java:252)
	at io.reactivex.internal.observers.DisposableLambdaObserver.dispose(DisposableLambdaObserver.java:90)
	at io.reactivex.internal.operators.observable.ObservableDoFinally$DoFinallyObserver.dispose(ObservableDoFinally.java:97)
	at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124)
	at io.reactivex.internal.operators.observable.ObservablePublishAlt$PublishConnection.dispose(ObservablePublishAlt.java:161)
	at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124)
	at io.reactivex.internal.operators.observable.ObservableRefCount.timeout(ObservableRefCount.java:156)
	at io.reactivex.internal.operators.observable.ObservableRefCount.cancel(ObservableRefCount.java:103)
	at io.reactivex.internal.operators.observable.ObservableRefCount$RefCountObserver.dispose(ObservableRefCount.java:252)
	at io.reactivex.internal.disposables.DisposableHelper.dispose(DisposableHelper.java:124)
	at io.reactivex.internal.observers.LambdaObserver.dispose(LambdaObserver.java:102)
	at com.kolibree.android.sdk.util.BluetoothUtilsImplTest.bluetoothStateObservable_disposeObserver_invokesonAllBluetoothStateObserversUnsubscribed(BluetoothUtilsImplTest.java:75)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.mockito.internal.junit.JUnitRule$1.evaluateSafely(JUnitRule.java:57)
	at org.mockito.internal.junit.JUnitRule$1.evaluate(JUnitRule.java:48)
	at com.kolibree.android.test.rules.ImmediateRxSchedulersOverrideRule$1.evaluate(ImmediateRxSchedulersOverrideRule.java:56)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.RuntimeException: Method getMainLooper in android.os.Looper not mocked. See http://g.co/androidstudio/not-mocked for details.
	at android.os.Looper.getMainLooper(Looper.java)
	at leakcanary.internal.InternalAppWatcher.<clinit>(InternalAppWatcher.kt:36)
	... 55 more
```

### Steps to Reproduce

I have this dependency in my build.gradle (it's a library, so I think I need to use implementation and not debugImplementation)

```implementation "com.squareup.leakcanary:leakcanary-object-watcher-android:2.0"```

I've tried to use `testImplementation libraries.leakCanaryWatcher` without success

**Expected behavior:** [What you expect to happen]

Unit tests pass

I guess it fails when creating the `InternalAppWatcher` singleton instance

```kotlin
private val mainHandler = Handler(Looper.getMainLooper())
```

Would this work?

```kotlin
private val mainHandler by lazy {
    Handler(Looper.getMainLooper())
}
```

`isInstalled` will then be able to check that `::application.isInitialized` is false

--- a/leakcanary-object-watcher-android/build.gradle
+++ b/leakcanary-object-watcher-android/build.gradle
@@ -5,6 +5,9 @@ dependencies {
   api project(':leakcanary-object-watcher')
 
   implementation deps.kotlin.stdlib
+
+  testImplementation deps.assertj_core
+  testImplementation deps.junit
 }
 
 android {

--- a/leakcanary-object-watcher-android/src/main/java/leakcanary/internal/InternalAppWatcher.kt
+++ b/leakcanary-object-watcher-android/src/main/java/leakcanary/internal/InternalAppWatcher.kt
@@ -14,6 +14,9 @@ import java.lang.reflect.InvocationHandler
 import java.lang.reflect.Proxy
 import java.util.concurrent.Executor
 
+/**
+ * Note: this object must load fine in a JUnit environment
+ */
 internal object InternalAppWatcher {
 
   val isInstalled
@@ -33,7 +36,9 @@ internal object InternalAppWatcher {
     }
   }
 
-  private val mainHandler = Handler(Looper.getMainLooper())
+  private val mainHandler by lazy {
+    Handler(Looper.getMainLooper())
+  }
 
   init {
     val internalLeakCanary = try {

--- a/leakcanary-object-watcher-android/src/test/java/leakcanary/AppWatcherTest.kt
+++ b/leakcanary-object-watcher-android/src/test/java/leakcanary/AppWatcherTest.kt
@@ -1,0 +1,13 @@
+package leakcanary
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+class AppWatcherTest {
+
+  @Test fun appWatcherLoads_notInstalled() {
+    assertThat(AppWatcher.isInstalled)
+        .describedAs("Ensure AppWatcher doesn't crash in JUnit tests")
+        .isFalse()
+  }
+}

